numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -4193,6 +4194,9 @@ normalize_axis_index(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
 }
 
 static struct PyMethodDef array_module_methods[] = {
+    {"_array_function_implementation_or_override",
```

Comment:
Name seems a bit long. What does `override` mean here that isn't implied by `_array_function_implementation`. Might consider the einsum approach where we have a `c_einsum`.

Reply:
Agreed. I renamed these to `implement_array_function` (I think `array_function_implementation` could be confused with the function `implementation` that gets passed into `implement_array_function`)